x-pack/legacy/plugins/ml/common/types/modules.ts

```diff
@@ -80,3 +80,5 @@ export interface DataRecognizerConfigResponse {
     dashboard: KibanaObjectResponse;
   };
 }
+
+export type JobOverride = Partial<Job>;
```

Comment:
Could `DeepPartial` we have in `x-pack/legacy/plugins/ml/common/types/common.ts` be of use here or are the overrides only optional on the most outer level?

Reply:
`DeepPartial` is messing up with arrays, for instance when you have an array of strings (`string[]`) it becomes `(string | undefined)[]`. Going to keep `Partial` 🙂